### PR TITLE
MISC: small fixes

### LINKED ISSUE #60: BN2 passive reputation when game closed
[North-West-Wind]
- [ ] Save file
- [ ] Minimal scripts to reproduce the issue
- [x] Steps to reproduce
- [x] Version of the game

## To Reproduce
Enter any faction on BN2. Close the game. Re-open the game after a while and see reputation in faction.

### More Information
Version: 1.6.4

I'm current doing a corporation run in BN2 and now in the endgame. Whenever I rejoin my game, I always get a few hundred millions reputation in Daedalus.

### LINKED ISSUE #85: Change documentation for Grow
[Walior]
I'm recommending few changes to the Growth documentation due to my confusion as a new player:

1) Add explanation of what happens if you grow a 0 money server to the documentation. 
Its not very intuitive and is fairly important. 

2) Change documentation for GrowthAnalyze to mention that it doesn't factor in the above mentioned effect. 

3) Change the wording of bitburner.ns.grow.md from

> the money available on a target server will be increased by a certain, fixed percentage 

 to

> the money available on a target server will be increased by a certain, fixed percentage of current money on the server 

in order to make the description more intuitive.

### LINKED ISSUE #265: Rep gained with SOA Does not match UI
Listed rep gain is exactly half of the actual gain. See screenshots:
image
image
image

Rep gain is shown as 3550.67 actual gain is 7102 ±1.
This is irrespective of faction favor, the WKS harmonizer, and other multipliers.

- [x ] Steps to reproduce
  - check SOA rep
  - do an infiltration
  - compare new SOA rep with amount gained in rewards screen
- [x ] Version of the game, e.g. Bitburner v2.1.0 (0133945c)

### PR DESCRIPTION
**Pull request by Mughur**
Actually fixing the servers going boinkers if your multipliers are too high by adding two more "if (x == infinity)x=1e300" checks

Gang respect could hit 0, making it impossible to gain any respect or money from gang. Fixed this by making the wanted penalty to use small value for respect if it somehow became 0.

Fixes: #265 `handleInfiltrators()` was call twice for each victory for some reason, giving twice the amount of reputation for SoA that the UI said

Fixes: #60

Closes: #85

### CODE REVIEW

File: src/ScriptEditor/NetscriptDefinitions.d.ts [hunk 1]
```diff
@@ -4786,6 +4787,8 @@ export interface NS {
    * the amount of money available on the specified server by the specified amount.
    * The specified amount is multiplicative and is in decimal form, not percentage.
    *
+   * Due to limitations of mathematics, this function won't the true value, but an approximation.
```
Comment (danielpza) [inline src/ScriptEditor/NetscriptDefinitions.d.ts]: possible typo?
```suggestion
   * Due to limitations of mathematics, this function won't be the true value, but an approximation.
```

Comment (Mughur) [reply src/ScriptEditor/NetscriptDefinitions.d.ts]: yup

**Comment by Snarling** [comment]
Test failure is fixed in dev and is unrelated to this PR.

**Comment by Zelow79** [comment]
Pretty sure this screwed up gangs.. 

https://github.com/bitburner-official/bitburner-src/commit/1baa615def7ecd3c20c33c99e2996f67a4d4849d#diff-7bc75fc68726fa582addfedbd0b9d90672dcebdf49cba06cd64271deb792394bL12

**Pull request merged**